### PR TITLE
propagate new bc::settings class

### PR DESCRIPTION
**Pull request by toxeus**
I'm planning to extend the parser once all constants have been moved into `bc::settings`.

### CODE REVIEW

**Comment by toxeus** [comment]
This was the last repo that needed an update for this certainly pervasive change.

File: include/bitcoin/explorer/command.hpp [hunk 1]
```diff
@@ -702,6 +702,14 @@ class BCX_API command
         setting_.server.client_private_key = value;
     }
 
+    /**
+     * Get the bitcoin settings.
+     */
+    virtual bc::settings& get_bitcoin_settings()
```
Comment (evoskuil) [inline include/bitcoin/explorer/command.hpp]: This is generated source. Each individual setting is generated from metadata in an xml file. The design here will need to be different.

Comment (toxeus) [reply include/bitcoin/explorer/command.hpp]: I see. I didn't pay attention to the comment above. I'm not familiar with gsl at all but I'll try to figure out the required changes by going through the existing gsl code in the model directory. @evoskuil please let me know if there's something important I would need to know when working with gsl.

**Comment by evoskuil** [comment]
You should not need to modify GSL code, but you need to look at the xml metadata and figure out how you are going to map in the settings. You will want to run the GSL script to observe the result of your changes.

**Comment by toxeus** [comment]
@evoskuil thanks. So [`generate.xml`](https://github.com/libbitcoin/libbitcoin-explorer/blob/master/model/generate.xml) is the metadata and [`generate.sh`](https://github.com/libbitcoin/libbitcoin-explorer/blob/master/model/generate.sh) is the GSL script. Is that correct?

**Comment by evoskuil** [comment]
Yes

**Comment by toxeus** [comment]
I've changed the code to use gsl now.

**Pull request merged**